servicetalk-client-api/src/main/java/io/servicetalk/client/api/DelayedRetry.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright © 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.client.api;
+
+import java.time.Duration;
+
+/**
+ * An interface that enhances any {@link Exception} to provide a constant {@link Duration delay} to be applied when
+ * it needs to be retried.
+ */
+public interface DelayedRetry {
+
+    /**
+     * A constant delay to apply.
+     *
+     * @return The {@link Duration} to apply as constant delay when retrying
+     */
+    Duration delay();
+
+    /**
+     * Returns original {@link Throwable} this {@link DelayedRetry} represents.
+     *
+     * @return the original {@link Throwable}
+     */
+    Throwable throwable();
```

Comment:
I found it useful for users of `retryDelayedRetryExceptions` function as a way to access exception and inspect its message/cause instead of casting the marker interface to `Throwable`. Lmk wdyt.

Reply:
Seems fine to me although it seems unhelpful for now considering the default implementation in `RetryingHttpRequesterFilter.DelayedRetry`.

Reply:
I had a case like that in `retryRetryableExceptions` function with `RetryableException` where I had to inspect the underlying exception type but it was not possible without a cast

Reply:
What I'm saying is that in the face of a default implementation that just throws it's dangerous to use that method so the safe thing to do is probably cast. You could make the default method attempt a downcast which should work for the intended use case:
```
@Override
        default Throwable throwable() {
            if (this instanceof Throwable) {
                return (Throwable) this;
            }
            throw new UnsupportedOperationException("DelayedRetry#throwable() is not supported by " + getClass());
        }
        ```